src/Planner/PlannerExpressionAnalysis.cpp

```diff
@@ -439,20 +441,30 @@ SortAnalysisResult analyzeSort(const QueryNode & query_node,
         auto & interpolate_list_node = query_node.getInterpolate()->as<ListNode &>();
 
         PlannerActionsVisitor interpolate_actions_visitor(planner_context);
-        auto interpolate_actions_dag = std::make_shared<ActionsDAG>();
+        auto interpolate_expression_dag = std::make_shared<ActionsDAG>();
 
         for (auto & interpolate_node : interpolate_list_node.getNodes())
         {
             auto & interpolate_node_typed = interpolate_node->as<InterpolateNode &>();
-            interpolate_actions_visitor.visit(interpolate_actions_dag, interpolate_node_typed.getExpression());
-            interpolate_actions_visitor.visit(interpolate_actions_dag, interpolate_node_typed.getInterpolateExpression());
+            interpolate_actions_visitor.visit(interpolate_expression_dag, interpolate_node_typed.getInterpolateExpression());
         }
 
         std::unordered_map<std::string_view, const ActionsDAG::Node *> before_sort_actions_inputs_name_to_node;
         for (const auto & node : before_sort_actions->getInputs())
             before_sort_actions_inputs_name_to_node.emplace(node->result_name, node);
 
-        for (const auto & node : interpolate_actions_dag->getNodes())
+        std::unordered_set<std::string_view> aggregation_keys;
+
+        auto projection_expression_dag = std::make_shared<ActionsDAG>();
+        for (const auto & node : query_node.getProjection())
+            actions_visitor.visit(projection_expression_dag, node);
+        for (const auto & node : projection_expression_dag->getNodes())
+            aggregation_keys.insert(node.result_name);
+
+        if (aggregation_analysis_result_optional)
+            aggregation_keys.insert(aggregation_analysis_result_optional->aggregation_keys.begin(), aggregation_analysis_result_optional->aggregation_keys.end());
```

Comment:
Why do we need it?

Reply:
for SELECT with aggregation in `INTERPOLATE` expression we only can use aggregated columns - here we assume that all columns in projection are aggregated (else it will throw exception on different stage of analysis), also we can use aggregated columns from `GROUP BY` expression - it can include some columns not present in projection.